### DISCUSSION TITLE
Zettel unique id filename reasoning?

### DISCUSSION THREAD

**Discussion by themaikimo** [bramses/bramses-highly-opinionated-vault-2023#9] [Q&A]
Hi. I have been around and around on whether to use the classic Zettel unique identifier as filename (of whatever format) or a more descriptive filename (or combo). 

I'm aware of several arguments in favor of each choice. 

Since this is a highly opinionated vault 😉 I wonder if you could share (or include in the README) your opinionated reasoning for choosing unique identifier-only filenames. 

(Btw, thank you for this vault and the super clear README. It's given me so many good ideas!)

**Comment by themaikimo** [comment]
Asking this question led me to this idea that counters my tendency to be all or nothing: Evan uses unique id filename for many/most incoming ideas, renames a few to claim phrase-filenamed evergreen note as clarity emerges.
http://evantravers.com/articles/2021/10/30/lifecycle-of-notes-my-implementation/

**Comment by bramses** [comment]
Thanks for reaching out @themaikimo! 

> your opinionated reasoning for choosing unique identifier-only filenames

The main reason is to never have to worry about filename conflicts with special characters. Many times in my file names I'd want to use characters like * or , or {}, etc. and all of those are disallowed from filenames (on UNIX systems at least).

Another benefit is that unique IDs as filenames can be easily inserted into a DB one to one if that ever comes up.

Finally, thanks to the Luhmann plugin (preinstalled in BHOV) you can search all valid Zettel ID files in a quick switcher like modal that will return the first h1 in the page: 

```
cmd + p -> Luhman: Open Zettel
```

> Notes containing a strong opinion or strong fact become a Named Note or Evergreen Note.

I really like this idea too though!

**Comment by bramses** [comment]
Jumping into this to say that UUIDs have proven immensely helpful in publishing to my blog, and never having to worry about how long slugs are, etc. The IDs are also very easy to backlink back to Obsidian and I don't have to worry about conflicts. https://github.com/bramses/obsidian-ghost-publish
